test/functional/test_files.py

```diff
@@ -288,18 +288,18 @@ def test_add_filepattern_from_dirname_recursive(client, cleanup_pins):
                     reason="No point in disabling os.fwalk if it isn't actually supported")
 def test_add_filepattern_from_dirname_recursive_nofwalk(client, cleanup_pins, monkeypatch):
 	monkeypatch.setattr(ipfshttpclient.filescanner, "HAVE_FWALK", False)
-	
+
 	res = client.add(FAKE_DIR_PATH, pattern=FAKE_DIR_FNPATTERN1, recursive=True)
 	assert conftest.sort_by_key(res) == conftest.sort_by_key(FAKE_DIR_FNPATTERN1_RECURSIVE_HASH)
 
 
-@pytest.mark.skipif(sys.platform.startswith("win"),
+@pytest.mark.skipif(not O_DIRECTORY,
```

Comment:
Please change this back, the presence of `O_DIRECTORY` has nothing to do with whether one can actually open directories on the given platform. All `O_DIRECTORY` does is *restricting* the given open call to *just* directories.

Reply:
I was trying to make the test restriction in terms of features rather than platform, in case the platform supports the feature in the future.

I've restored the original guard and added some comments in a pending change.